### PR TITLE
Add IsSolutionFullyOpen to ISolutionInfoProvider.

### PR DESCRIPTION
**Pull request by bigfluffycookie**
fixes https://github.com/SonarSource/sonarlint-visualstudio/issues/4760

### CODE REVIEW

File: src/Core/ISolutionInfoProvider.cs [hunk 1]
```diff
@@ -54,5 +54,18 @@ public interface ISolutionInfoProvider
         /// <remarks>If possible, use the asynchronous version of this method. This synchronous version was added
         /// to be used only by existing non-thread-aware code that cannot easily be refactored.</remarks>
         string GetFullSolutionFilePath();
+
```
Comment (bigfluffycookie) [inline src/Core/ISolutionInfoProvider.cs:57]: Removing this with other requested changes

File: src/Infrastructure.VS.UnitTests/SolutionInfoProviderTests.cs [hunk 2]
```diff
@@ -67,7 +67,7 @@ public void Ctor_DoesNotCallServices()
         [DataRow("a value")]
         public async Task GetSlnFilePathAsync_ReturnsExpectedValue(string solutionNameToReturn)
         {
-            var solution = CreateIVsSolution(solutionNameToReturn);
+            var solution = CreateIVsSolutionWithSolutionFileName(solutionNameToReturn);
```
Comment (bigfluffycookie) [inline src/Infrastructure.VS.UnitTests/SolutionInfoProviderTests.cs:70]: Renamed it as we now need two different createSolution methods.

File: src/Infrastructure.VS.UnitTests/SolutionInfoProviderTests.cs [hunk 3]
```diff
@@ -165,12 +209,35 @@ public async Task GetDirectoryAsync_ServiceCalledOnUIThread()
             calls.Should().ContainInOrder("switch to UI thread", "GetService", "GetSolutionName");
         }
 
+        [TestMethod]
```
Comment (bigfluffycookie) [inline src/Infrastructure.VS.UnitTests/SolutionInfoProviderTests.cs:212]: The new tests here are pretty similar to the existing tests.

File: src/Integration/LocalServices/IProjectSystemHelper.cs [hunk 4]
```diff
@@ -84,10 +84,5 @@ public interface IProjectSystemHelper : IVsHierarchyLocator
         /// </summary>
         /// <returns>Project kinds GUIDs for the project</returns>
         IEnumerable<Guid> GetAggregateProjectKinds(IVsHierarchy hierarchy);
-
-        /// <summary>
-        /// Returns a flag indicating whether there is fully-opened solution or not.
-        /// </summary>
-        bool IsSolutionFullyOpened();
```
Comment (duncanp-sonar) [inline src/Integration/LocalServices/IProjectSystemHelper.cs:91]: Is there an implementation of the method in `ConfigurableVsProjectSystemHelper` that can be removed?

**Review by duncanp-sonar — APPROVED** [review]

**Pull request merged**